cmake/StandardWarnings.cmake

```diff
@@ -137,4 +137,13 @@ elseif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
     add_target_compile_flag_if_supported(
       standard_warnings "-Wno-unused-variable")
   endif()
+elseif(MSVC)
```

Comment:
for followup commits consider e.g. W4

Reply:
Will do!

Reply:
I added `/W4`, now there are much more warnings, but still less than one screenful.

Reply:
And now I silenced all the warnings, none of them were worth keeping.